crates/ruff/src/rules/flake8_type_checking/rules/typing_only_runtime_import.rs

```diff
@@ -342,6 +368,42 @@ pub(crate) fn typing_only_runtime_import(
             }
         };
 
+        if checker.patch(diagnostic.kind.rule()) {
+            diagnostic.try_set_fix(|| {
+                // Step 1) Remove the import.
+                let source = binding.source.unwrap();
+                let deleted: Vec<&Stmt> = checker.deletions.iter().map(Into::into).collect();
```

Comment:
This code looks (identical?) to your other PR. Is there an opportunity to share some of the logic?

Reply:
We have these everywhere that we do statement deletion :(